app/src/main/java/org/simple/clinic/widgets/ageanddateofbirth/DateOfBirthFormatValidator.kt

```diff
@@ -17,27 +17,42 @@ class DateOfBirthFormatValidator @Inject constructor(
     DATE_IS_IN_FUTURE,
   }
 
+  sealed class Result2 {
```

Comment:
Instead of adding `Result2`, `validate2` methods in the same class, what do you think about creating another class `DateOfBithFormatValidator2` that then has nicely named methods? The older validator could get a reference to the new one via injection and classes could inject the newer class easily.

Reply:
I don't see the benefit 🤔 

> that then has nicely named methods

what do you mean by this?

Reply:
One benefit of keeping both the functions in the same class is discoverability during code completion.

Reply:
> I don't see the benefit 🤔
> 
> > that then has nicely named methods
> 
> what do you mean by this?

What I mean is there won't be a `Result2`, `validate2`, etc. This will only be used when declaring the type `DateOfBirthFormatValidator2` at the injection site. The actual usages of the instance will be cleaner, like

```kotlin
dateValidator.validate()
```

However, we will be removing this soon anyway once we replace the deprecated usages with the newer ones. We can leave this in for now, and fix the method naming once the migration completes.